pkg/schedule/filter/affinity.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2025 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filter
+
+import (
+	"strconv"
+
+	"github.com/pingcap/kvproto/pkg/pdpb"
+
+	"github.com/tikv/pd/pkg/core"
+	"github.com/tikv/pd/pkg/schedule/affinity"
+	"github.com/tikv/pd/pkg/schedule/config"
+	sche "github.com/tikv/pd/pkg/schedule/core"
+	"github.com/tikv/pd/pkg/schedule/plan"
+)
+
+type affinityFilter struct {
+	affinityManager *affinity.Manager
+}
+
+// NewAffinityFilter creates a RegionFilter that filters all affinity regions.
+func NewAffinityFilter(cluster sche.SharedCluster) RegionFilter {
+	return &affinityFilter{
+		affinityManager: cluster.GetAffinityManager(),
+	}
+}
+
+// Select implements the RegionFilter interface.
+func (f *affinityFilter) Select(region *core.RegionInfo) *plan.Status {
+	if f.affinityManager != nil {
+		group, _ := f.affinityManager.GetRegionAffinityGroupState(region)
+		if group != nil && !group.RegularSchedulingEnabled {
+			return statusRegionAffinity
+		}
+	}
+	return statusOK
+}
+
+// For affinity regions, we use calculation: (maxSize + buffer) * multiplier
+// Derived from default values:
+//   - Default MaxMergeRegionSize = 54MB
+//   - Default region-split-size = 256MB
+const (
+	affinityRegionSizeBufferMB   = 10
+	affinityRegionSizeMultiplier = 4
+)
+
+// AllowAutoSplit returns true if the region can be auto split
+func AllowAutoSplit(cluster sche.ClusterInformer, region *core.RegionInfo, reason pdpb.SplitReason) bool {
+	if region == nil {
+		// The default behavior is to allow it.
+		return true
+	}
+
+	if !cluster.GetCheckerConfig().IsAffinitySchedulingEnabled() {
+		return true
+	}
+
+	switch reason {
```

Comment:
if reason == SplitReason_ADMIN{
    return true 
}
It just check the reason is admin, not needs to use switch

Reply:
ok